src/WriteStreamCSV.h

```diff
@@ -66,10 +66,11 @@ class WriteFileCSV : public WriteStreamCSV, public WriteStream {
     }
 
     void writeNextTuple(const RamDomain* tuple) override {
-        writeValue(file, 0, tuple[0]);
-        for (size_t col = 1; col < arity; ++col) {
-            file << delimiter;
-            writeValue(file, col, tuple[col]);
+        for (size_t col = 0; col < arity; ++col) {
+            writeValue(file, tuple[col], typeMask.at(col));
+            if (col != arity - 1) {
```

Comment:
If you write the first value before the loop, then loop from 1 to end, the if statement is not needed and the (entirely safe right now) check of `unsigned value - 1` is avoided.